### ISSUE TITLE
Export with dependencies is missing autoload files

### ISSUE THREAD

**Issue by firegate666** [godotengine/godot#32377]
**Godot version:** 3.1.1

**OS/device including version:** OS X 10.14.6 (18G95)

**Issue description:** When I select export selected scenes and dependencies, it misses the files declared as autoloaded which renders the game unplayable.

Even other scenes that are directly inside a scene are missing: Player.tscn

**Steps to reproduce:** Have in autoloader and export selected scenes and dependencies

**Project:** https://github.com/firegate666/godot-platformer

```
ERROR: initialize: Directional shadow framebuffer status invalid
   At: drivers/gles2/rasterizer_scene_gles2.cpp:3369.
ERROR: load_interactive: Condition ' err != OK ' is true. returned: Ref<ResourceInteractiveLoader>()
   At: scene/resources/resource_format_text.cpp:1226.
ERROR: _load: Failed loading resource: res://default_env.tres
   At: core/io/resource_loader.cpp:285.
ERROR: SceneTree: Default Environment as specified in Project Settings (Rendering -> Environment -> Default Environment) could not be loaded.
   At: scene/main/scene_tree.cpp:2048.
ERROR: load_source_code: Condition ' err ' is true. returned: err
   At: modules/gdscript/gdscript.cpp:813.
ERROR: load: Condition ' err != OK ' is true. returned: RES()
   At: modules/gdscript/gdscript.cpp:2172.
ERROR: _load: Failed loading resource: res://SignalBus.gd
   At: core/io/resource_loader.cpp:285.
ERROR: start: Can't autoload: res://SignalBus.gd
   At: main/main.cpp:1549.
ERROR: load_source_code: Condition ' err ' is true. returned: err
   At: modules/gdscript/gdscript.cpp:813.
ERROR: load: Condition ' err != OK ' is true. returned: RES()
   At: modules/gdscript/gdscript.cpp:2172.
ERROR: _load: Failed loading resource: res://Player.gd
   At: core/io/resource_loader.cpp:285.
ERROR: poll: res://Player.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://Player.gd
   At: scene/resources/resource_format_text.cpp:440.
ERROR: load: Condition ' err != OK ' is true. returned: RES()
   At: core/io/resource_loader.cpp:208.
ERROR: _load: Failed loading resource: res://Player.tscn
   At: core/io/resource_loader.cpp:285.
ERROR: poll: res://Level1.tscn:4 - Parse Error: [ext_resource] referenced nonexistent resource at: res://Player.tscn
   At: scene/resources/resource_format_text.cpp:440.
ERROR: load: Condition ' err != OK ' is true. returned: RES()
   At: core/io/resource_loader.cpp:208.
ERROR: _load: Failed loading resource: res://Level1.tscn
   At: core/io/resource_loader.cpp:285.

```

**Comment by qarmin** [comment]
Did you try to change Autoload or class_name - PlayerModel?
In Godot 3.2+ there is visible error when someone have same name for Autoload and Class.
```
SCRIPT ERROR: GDScript::reload: Parse Error: The class "PlayerModel" conflicts with the AutoLoad singleton of the same name, and is therefore redundant. Remove the class_name declaration to fix this error.
   At: res://PlayerModel.gd:2.
ERROR: reload: Method/Function Failed, returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:576.
ERROR: _create_autoload: Script does not inherit a Node: res://PlayerModel.gd.
   At: editor/editor_autoload_settings.cpp:327.
```

**Comment by RokaRoka** [comment]
I've been having this issue recently. My temporary fix has been putting all my autoloads in a folder (e.g. res://AutoLoads/), then in the export settings, write the folder as a filter to export "non-resource files/folders" (e.g. write AutoLoads/*).

Though this works, I think it would be nice to have the option in export to include or exclude AutoLoaded files when exporting with scenes + dependencies :)

**Comment by jkb0o** [comment]
There is not only autoload but also global classes (`class_name` used in selected scenes/scripts) are missed in export.

**Comment by fmazan** [comment]
Confirmed, this still happens with autoload scripts in Godot Engine v3.2.2.stable.official and Android export template.

**Comment by fmazan** [comment]
I can fix this issue, as I have looked at the code and it's quite straightforward. The question is though, which of these options would be the best, @KoBeWi?

1. automatically export autoloaded resources and their dependencies no matter what
2. extend ExportFilter enum by adding EXPORT_SELECTED_SCENES_AND_AUTOLOAD, and hence add a new option item "Export selected scenes and AutoLoad (and dependencies)"
3. in case of EXPORT_SELECTED_SCENES is selected, show additional CheckBox in the ProjectExportDialog right under the OptionButton determining whether to export autoload resources or not
4. leave it as-is, and use workaround @RokaRoka mentioned

**Comment by KoBeWi** [comment]
I think 1. would be the best, because it fixes the bug and is straightforward to do. We could make it customizable later if someone wants it (but hard to say right now, likely not).

**Comment by akien-mga** [comment]
> There is not only autoload but also global classes (`class_name` used in selected scenes/scripts) are missed in export.

Could you open a dedicated issue for this? The autoload part will be fixed by #45059.

**Comment by exodrifter** [comment]
I opened #111422 and #111423 to address the missing classes when exporting selected scenes and dependencies.
